test/api_callable.js

```diff
@@ -523,6 +523,29 @@ describe('bundleable', function() {
         .then(bundledCallback)
         .catch(done);
   });
+
+  it('cancels partially on bundling method', function(done) {
+    var apiCall = createApiCall(func, settings);
+    var expectedSuccess = false;
+    var expectedFailure = false;
+    apiCall(createRequest([1, 2, 3], 'id'), null).then(function(obj) {
+      expect(obj.field1).to.deep.equal([1, 2, 3]);
+      expectedSuccess = true;
+      if (expectedSuccess && expectedFailure) {
+        done();
+      }
+    }).catch(done);
+    var p = apiCall(createRequest([1, 2, 3], 'id'), null);
+    p.then(function(obj) {
+      done(new Error('should not succeed'));
+    }).catch(function(err) {
+      expectedFailure = true;
+      if (expectedSuccess && expectedFailure) {
+        done();
+      }
+    });
```

Comment:
I think this test is a tiny bit flaky since there are no guarantees that cancel will run before the delay threshold is passed (although most of the time it should). Add a todo here to add time mocking when we end up fixing the other tests that need time mocking.